api/pod-handlers.go

```diff
@@ -262,77 +268,83 @@ func getDescribePodResponse(session *models.Principal, params operator_api.Descr
 		statusKeys[i] = status.Name
 
 	}
-	for i := range pod.Spec.Containers {
+	for _, container := range pod.Spec.Containers {
 		retval.Containers[i] = &models.Container{
-			Name:      pod.Spec.Containers[i].Name,
-			Image:     pod.Spec.Containers[i].Image,
-			Ports:     describeContainerPorts(pod.Spec.Containers[i].Ports),
-			HostPorts: describeContainerHostPorts(pod.Spec.Containers[i].Ports),
-			Args:      pod.Spec.Containers[i].Args,
+			Name:      container.Name,
+			Image:     container.Image,
+			Ports:     describeContainerPorts(container.Ports),
+			HostPorts: describeContainerHostPorts(container.Ports),
+			Args:      container.Args,
 		}
-		if slices.Contains(statusKeys, pod.Spec.Containers[i].Name) {
-			retval.Containers[i].ContainerID = statusMap[pod.Spec.Containers[i].Name].ContainerID
-			retval.Containers[i].ImageID = statusMap[pod.Spec.Containers[i].Name].ImageID
-			retval.Containers[i].Ready = statusMap[pod.Spec.Containers[i].Name].Ready
-			retval.Containers[i].RestartCount = int64(statusMap[pod.Spec.Containers[i].Name].RestartCount)
-			retval.Containers[i].State, retval.Containers[i].LastState = describeStatus(statusMap[pod.Spec.Containers[i].Name])
+		if slices.Contains(statusKeys, container.Name) {
+			containerStatus := statusMap[container.Name]
+			retval.Containers[i].ContainerID = containerStatus.ContainerID
+			retval.Containers[i].ImageID = containerStatus.ImageID
+			retval.Containers[i].Ready = containerStatus.Ready
+			retval.Containers[i].RestartCount = int64(containerStatus.RestartCount)
+			retval.Containers[i].State = describeContainerState(containerStatus.State)
+			retval.Containers[i].LastState = describeContainerState(containerStatus.LastTerminationState)
 		}
-		retval.Containers[i].EnvironmentVariables = make([]*models.EnvironmentVariable, len(pod.Spec.Containers[i].Env))
-		for j := range pod.Spec.Containers[i].Env {
+		retval.Containers[i].EnvironmentVariables = make([]*models.EnvironmentVariable, len(container.Env))
+		for j, env := range container.Env {
 			retval.Containers[i].EnvironmentVariables[j] = &models.EnvironmentVariable{
-				Key:   pod.Spec.Containers[i].Env[j].Name,
-				Value: pod.Spec.Containers[i].Env[j].Value,
+				Key:   env.Name,
+				Value: env.Value,
 			}
 		}
-		retval.Containers[i].Mounts = make([]*models.Mount, len(pod.Spec.Containers[i].VolumeMounts))
-		for j := range pod.Spec.Containers[i].VolumeMounts {
+		retval.Containers[i].Mounts = make([]*models.Mount, len(container.VolumeMounts))
+		for j, volumeMount := range container.VolumeMounts {
 			retval.Containers[i].Mounts[j] = &models.Mount{
```

Comment:
we could use append instead thoughts?

Reply:
I missed `retval.Containers` is using the variable `i` :P now  `i` is not being initiliazed in the loop, that will break things.
Current code looks complex so probably it might be better to revert to your previous commit for now

Reply:
I just saw it lol, it already started breaking things. I agree.

Reply:
reverted